components/TextInput.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { Controller, Control } from 'react-hook-form';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import { FormValues } from '@/types';
+
+type TextInputProps = TextFieldProps &  {
+  name: keyof FormValues;
+  label: string;
```

Comment:
1. Do we need `label` and `fullWidth`? They should be included in `TextFieldProps` already. 
2. I would recommend you to manage input widths from the `page.tsx` file by wrapping inputs with `Grid`. If I ever want to use `1/3` or `1/4` width input, then by your example I would need to add additional props to each input type. Not flexible.